bson/src/main/org/bson/json/UuidStringValidator.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.json;
+
+import java.util.BitSet;
+
+final class UuidStringValidator {
+    private static final BitSet HEX_CHARS;
+
+    static {
+        BitSet hexChars = new BitSet(256);
+
+        hexChars.set('0', '9' + 1);
+        hexChars.set('A', 'F' + 1);
+        hexChars.set('a', 'f' + 1);
+
+        HEX_CHARS = hexChars;
```

Comment:
@stIncMale the Javadoc for BitSet states that it's not safe for multithreaded use without external synchronization.  But given that the use is read-only after assignment to a static final field, I think that is sufficient synchronization.  Is that your understanding as well?

Reply:
Actually, I don't see anything in https://docs.oracle.com/javase/specs/jls/se15/html/jls-17.html#jls-17.4.5 that indicates that this is safe.

Reply:
Initializing static state in a static initializer and only reading it outside of the initializer is thread-safe: roughly speaking, all write actions produced by a static initializer happen-before reads (and writes, but that is irrelevant for this specific situation) produced by the code outside of it.

Unfortunately, there is no note in the JMM stating this, and we are left figuring out on our own from different places in the JLS. The above statement follows from [JLS 12.4.1. When Initialization Occurs](https://docs.oracle.com/javase/specs/jls/se15/html/jls-12.html#jls-12.4.1), [12.4.2. Detailed Initialization Procedure](https://docs.oracle.com/javase/specs/jls/se15/html/jls-12.html#jls-12.4.2),

Reply:
Just in case: having a local variable `hexChars` for the purpose of modifying the `BitSet` state and then writing `HEX_CHARS = hexChars` does not change anything (the same is true for `final` fields because a [_"freeze action on `final` field f of `o` takes place when `c` (constructor) exits, either normally or abruptly"_](https://docs.oracle.com/javase/specs/jls/se15/html/jls-17.html#jls-17.5.1)). So we can just have the following code:
```java
static {
    HEX_CHARS = new BitSet(256);
    HEX_CHARS.set('0', '9' + 1);
    HEX_CHARS.set('A', 'F' + 1);
    HEX_CHARS.set('a', 'f' + 1);
}
```

Reply:
I implemented your suggested change